### DISCUSSION TITLE
Left join of table

### DISCUSSION THREAD

**Discussion by shivarajmulimani** [cozodb/cozo#100] [Q&A]
Given the common key in two tables how to make left and right joins?
Any examples please?

**Comment by zh217** [comment]
You have to consider the two cases separately:
```
a[x, y] <- [[1, 2], [3, 4]]
b[y, z] <- [[2, 3], [2, 4]]
?[x, y, z] := a[x, y], b[y, z]
?[x, y, z] := a[x, y], not b[y, _], z = null
```

This should give:
```
1	2	3
1	2	4
3	4	null
```

**Comment by shivarajmulimani** [comment]
if I understood it correctly 
?[x, y, z] := a[x, y], b[y, z] this section is joining wherever there is match of y in b.

the below part 
?[x, y, z] := a[x, y], not b[y, _], z = null , here you are taking where there is no match of y in b.
But I didn't understand what is "_" variable  doing,  can you explain the meaning of it - "?[x, y, z] := a[x, y], not b[y, _], z = null"

**Reply by zh217** [reply]
the underscore just creates a unique variable that is never unified even with itself when there are multiple instances. you can use any other variable in this case as long as it is not x, y or z.
